modeltranslation/thread_context.py

```diff
@@ -0,0 +1,42 @@
+import threading
+
+from modeltranslation import settings
+
+
+class ModelTranslationThreadLocal(threading.local):
+    """Holds thread-local data for modeltranslation."""
+    auto_populate = None
```

Comment:
maybe it's better to set default values from settings?

```suggestion
    auto_populate: Union[bool, Literal['all', 'default', 'required']] = settings.AUTO_POPULATE
```

so we can remove "is not None" checks from getters.

And let's try to provide types for new code.

Reply:
@last-partizan Not setting default value was intensional here. Setting default value will make `fallbacks_enabled()` and `auto_populate_mode()` always return the value from thread local (unless context manager is used). And this will break in 2 places:

1. In tests `seetings.override` will not work as expected for `ENABLE_FALLBACKS` and `AUTO_POPULATE`.
2. if for some reason user changes theses settings at run time the thread locals value will not change. So the settings change will not be effective.

Please let me know if this makes sense. :)

Reply:
Similar to:
https://github.com/django/django/blob/967f8750ab63f9ca74ce63ada580ccc5b10b3f3b/django/utils/translation/trans_real.py#L320

https://github.com/django/django/blob/967f8750ab63f9ca74ce63ada580ccc5b10b3f3b/django/utils/translation/trans_real.py#L301